### ISSUE TITLE
ACRA 4.8.5 Cannot initiate CrashReportDialog class

### ISSUE THREAD

**Issue by CKDPublishing** [ACRA/acra#423]
I tried ACRA 4.8.5 with annotation and got the following error. Note: You can ignore the first part of the error as this part was done deliberately to crash the app so it can launch the ACRA error dialog:

`
04-07 23:20:11.189: E/ACRA(8256): ACRA caught a RuntimeException for au.com.company.test
04-07 23:20:11.189: E/ACRA(8256): java.lang.RuntimeException: Unable to start activity ComponentInfo{au.com.company.test/au.com.company.test.MainActivity}: java.lang.NumberFormatException: Invalid int: "null"
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2658)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2723)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.access$900(ActivityThread.java:172)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1422)
04-07 23:20:11.189: E/ACRA(8256): 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-07 23:20:11.189: E/ACRA(8256): 	at android.os.Looper.loop(Looper.java:145)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.main(ActivityThread.java:5832)
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.reflect.Method.invoke(Native Method)
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.reflect.Method.invoke(Method.java:372)
04-07 23:20:11.189: E/ACRA(8256): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
04-07 23:20:11.189: E/ACRA(8256): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
04-07 23:20:11.189: E/ACRA(8256): Caused by: java.lang.NumberFormatException: Invalid int: "null"
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.Integer.invalidInt(Integer.java:138)
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.Integer.parseInt(Integer.java:358)
04-07 23:20:11.189: E/ACRA(8256): 	at java.lang.Integer.parseInt(Integer.java:334)
04-07 23:20:11.189: E/ACRA(8256): 	at au.com.company.test.MainActivity.onCreate(MainActivity.java:114)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.Activity.performCreate(Activity.java:6221)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
04-07 23:20:11.189: E/ACRA(8256): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2611)
04-07 23:20:11.189: E/ACRA(8256): 	... 10 more
04-07 23:20:11.349: E/ACRA(8256): ACRA failed to capture the error - handing off to native error reporter
04-07 23:20:11.349: E/ACRA(8256): android.content.ActivityNotFoundException: Unable to find explicit activity class {au.com.company.test/org.acra.dialog.CrashReportDialog}; have you declared this activity in your AndroidManifest.xml?
04-07 23:20:11.349: E/ACRA(8256): 	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:1777)
04-07 23:20:11.349: E/ACRA(8256): 	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1499)
04-07 23:20:11.349: E/ACRA(8256): 	at android.app.ContextImpl.startActivity(ContextImpl.java:1545)
04-07 23:20:11.349: E/ACRA(8256): 	at android.app.ContextImpl.startActivity(ContextImpl.java:1527)
04-07 23:20:11.349: E/ACRA(8256): 	at android.content.ContextWrapper.startActivity(ContextWrapper.java:337)
04-07 23:20:11.349: E/ACRA(8256): 	at org.acra.builder.ReportExecutor.dialogAndEnd(ReportExecutor.java:234)
04-07 23:20:11.349: E/ACRA(8256): 	at org.acra.builder.ReportExecutor.execute(ReportExecutor.java:222)
04-07 23:20:11.349: E/ACRA(8256): 	at org.acra.builder.ReportBuilder.build(ReportBuilder.java:148)
04-07 23:20:11.349: E/ACRA(8256): 	at org.acra.ErrorReporter.uncaughtException(ErrorReporter.java:260)
04-07 23:20:11.349: E/ACRA(8256): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:693)
04-07 23:20:11.349: E/ACRA(8256): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:690)
04-07 23:20:11.349: E/AndroidRuntime(8256): FATAL EXCEPTION: main
04-07 23:20:11.349: E/AndroidRuntime(8256): Process: au.com.company.test, PID: 8256
04-07 23:20:11.349: E/AndroidRuntime(8256): java.lang.RuntimeException: Unable to start activity ComponentInfo{au.com.company.test/au.com.company.test.MainActivity}: java.lang.NumberFormatException: Invalid int: "null"
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2658)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2723)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.access$900(ActivityThread.java:172)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1422)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.os.Looper.loop(Looper.java:145)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.main(ActivityThread.java:5832)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.reflect.Method.invoke(Native Method)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.reflect.Method.invoke(Method.java:372)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1399)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1194)
04-07 23:20:11.349: E/AndroidRuntime(8256): Caused by: java.lang.NumberFormatException: Invalid int: "null"
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.Integer.invalidInt(Integer.java:138)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.Integer.parseInt(Integer.java:358)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at java.lang.Integer.parseInt(Integer.java:334)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at au.com.company.test.MainActivity.onCreate(MainActivity.java:114)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.Activity.performCreate(Activity.java:6221)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2611)
04-07 23:20:11.349: E/AndroidRuntime(8256): 	... 10 more
 `

The interesting bit is this:

`android.content.ActivityNotFoundException: Unable to find explicit activity class {au.com.company.test/org.acra.dialog.CrashReportDialog}; have you declared this activity in your AndroidManifest.xml?`

Even though I have declared the following directives in my AndroidManifest.xml file:

```
<activity
    android:name="org.acra.CrashReportDialog"
    android:theme="@style/Theme.Dialog"
    android:launchMode="singleInstance"
    android:excludeFromRecents="true"
    android:finishOnTaskLaunch="true">
</activity>
<service
    android:name="org.acra.sender.SenderService"
    android:exported="false"
    android:process=":acra"/>
```

So, is the error complaining about the missing directive in the ACRA 4.8.5.jar itself?

**Comment by william-ferguson-au** [comment]
You have defined an Activity of 'org.acra.CrashReportDialog'

But by default ACRA is looking for 'org.acra.dialog.CrashReportDialog'

Note the difference in package name. If you are using manifest merging you don't need to define either of those components as they are already defining in the ACRA AAR.

**Comment by CKDPublishing** [comment]
> You have defined an Activity of 'org.acra.CrashReportDialog'
> But by default ACRA is looking for 'org.acra.dialog.CrashReportDialog'

Thanks, William. Can't believe I miss the obvious.

Anyway, I have changed the directive to 'org.acra.dialog.CrashReportDialog'. Here's a couple of tests that I ran:

- With the directives specified in my app's own 'AndroidManifest' file, I get the crash dialog window. But clicking the OK button in the crash dialog window simply close the window without going to an email client so the error can be sent via a custom 'ReportSender'.

- Comment out the directives as you suggested, I just got the default crash window from Android instead of the one from ACRA. I suppose this behaviour is related to this ACRA [issue](https://github.com/ACRA/acra/issues/405).
